posthog/api/insight.py

```diff
@@ -135,13 +143,24 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Any:
 
     def to_representation(self, instance):
         representation = super().to_representation(instance)
+
+        # the ORM doesn't know about deleted dashboard tiles when filling dashboards relation
+        # use the dashboard_tiles set to filter them
+        representation["dashboards"] = [
+            id for id in representation["dashboards"] if id in self._dashboard_tiles(instance)
+        ]
+
         filters = instance.dashboard_filters()
 
         if not filters.get("date_from"):
             filters.update({"date_from": f"-{DEFAULT_DATE_FROM_DAYS}d"})
         representation["filters"] = filters
         return representation
 
+    @lru_cache(maxsize=1)
+    def _dashboard_tiles(self, instance):
+        return [tile.dashboard_id for tile in instance.dashboard_tiles.all()]
```

Comment:
Should we be filtering out the soft-deleted ones here?

Reply:
Yep, that adds a query per insight 🤷 . The Django ORM is taunting me at the moment